src/sia/services/response_handler.py

```diff
@@ -0,0 +1,103 @@
+"""Module for the Query Processor service."""
+
+from collections.abc import Callable
+from datetime import UTC, datetime
+
+import astropy
+import structlog
+from fastapi import Request
+from lsst.daf.butler import Butler
+from lsst.dax.obscore import ExporterConfig
+from lsst.dax.obscore.siav2 import SIAv2Parameters
+from starlette.responses import Response
+
+from ..constants import RESULT_NAME as RESULT
+from ..factory import Factory
+from ..models.sia_query_params import SIAQueryParams
+from ..services.votable import VotableConverterService
+
+logger = structlog.get_logger(__name__)
+
+SIAv2QueryType = Callable[
+    [Butler, ExporterConfig, SIAv2Parameters],
+    astropy.io.votable.tree.VOTableFile,
+]
+
+
+class ResponseHandlerService:
+    """Service for handling the SIAv2 query response."""
+
+    @staticmethod
+    def process_query(
+        *,
+        factory: Factory,
+        params: SIAQueryParams,
+        sia_query: SIAv2QueryType,
+        request: Request,
+        token: str | None,
+    ) -> Response:
+        """Process the SIAv2 query and generate a Response.
+
+        Parameters
+        ----------
+        factory
+            The Factory instance.
+        params
+            The parameters for the SIAv2 query.
+        sia_query
+            The SIA query method to use
+        request
+            The request object.
+        token
+            The token to use for the Butler (Optional).
+
+        Returns
+        -------
+        Response
+            The response containing the query results.
+        """
+        logger.info(
+            "SIA query started with params:",
+            params=params,
+            method=request.method,
+            time=datetime.now(UTC).strftime("%Y-%m-%d %H:%M:%S"),
```

Comment:
In general you shouldn't add the current time to log parameters because both structlog and Kubernetes can do this for you. I usually just rely on the Kubernetes log timestamps, but if you really want explicit timings in the JSON log, you can use the `add_timestamp` argument to `configure_logging`.